code/scripting/api/objs/model.cpp

```diff
@@ -737,52 +716,37 @@ ADE_OBJ(l_ModelThrusters, model_h, "thrusters", "The thrusters of a model");
 
 ADE_FUNC(__len, l_ModelThrusters, NULL, "Number of thruster banks on the model", "number", "Number of thrusterbanks")
 {
-	model_h *trh;
-	if(!ade_get_args(L, "o", l_ModelThrusters.GetPtr(&trh)))
-		return ade_set_error(L, "i", -1);
-
-	if(!trh->isValid())
+	model_h *mdl;
+	if (!ade_get_args(L, "o", l_ModelThrusters.GetPtr(&mdl)))
 		return ade_set_error(L, "i", -1);
 
-	polymodel *pm = trh->Get();
-
-	if(pm == NULL)
+	polymodel *pm = mdl->Get();
+	if (!pm)
 		return ade_set_error(L, "i", -1);
 
 	return ade_set_args(L, "i", pm->n_thrusters);
 }
 
 ADE_INDEXER(l_ModelThrusters, "number Index", "Array of all thrusterbanks on this thruster", "thrusterbank", "Handle to the thrusterbank or invalid handle if index is invalid")
 {
-	model_h *trh = NULL;
-	const char* s    = nullptr;
-	thrusterbank_h newThr;
+	model_h *mdl = nullptr;
+	int idx = -1;
 
-	if (!ade_get_args(L, "os|o", l_ModelThrusters.GetPtr(&trh), &s, l_Thrusterbank.Get(&newThr)))
+	if (!ade_get_args(L, "oi", l_ModelThrusters.GetPtr(&mdl), &idx))
```

Comment:
Will this potentially break things?

Reply:
No, for all practical purposes this has always behaved as a numeric indexer.  The old code did not index by string (e.g. as a name); what it did was load the argument as a string and then use `atoi` to interpret it as a number.  The new code just loads the argument as a number.